plinth/modules/first_boot/middleware.py

```diff
@@ -37,19 +38,55 @@ class FirstBootMiddleware(object):
     def process_request(request):
         """Handle a request as Django middleware request handler."""
         state = kvstore.get_default('firstboot_state', 0)
-
-        firstboot_index_url = reverse('first_boot:index')
-        user_requests_firstboot = request.path.startswith(firstboot_index_url)
-
+        user_requests_firstboot = is_firstboot(request.path)
+        user_requests_login = request.path.startswith(reverse(settings.LOGIN_URL))
         help_index_url = reverse('help:index')
         user_requests_help = request.path.startswith(help_index_url)
+        if not user_requests_login and not user_requests_help:
+            if state == 1 and user_requests_firstboot:
```

Comment:
I think this should be "state > 0". (Or perhaps, we can convert 'firstboot_state' to boolean?)

For existing plinth installs, most likely the state is 10, since firstboot has been completed up to that point. If we only check for state == 1, then someone can access /plinth/firstboot and run firstboot again.